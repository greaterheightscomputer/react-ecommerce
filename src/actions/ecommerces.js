import uuid from 'uuid';

//Add_Ecommerce
export const addEcommerce = ({ description='', category='', item='', amount=0, image='', imageUrl='', stock=0, createdAt=0 } = {}) => ({
    type: 'ADD_ECOMMERCE',
    ecommerce: {
        id: uuid(),        
        description,
        category,
        item,
        amount,
        image,
        imageUrl,
        stock,
        createdAt
    }
});
// REMOVE_Ecommerce    
export const removeEcommerce = ( { id } = {}) => ({
    type: 'REMOVE_ECOMMERCE',
    id
});
// EDIT_Ecommerce  
export const editEcommerce = (id, updates) => ({
    type: 'EDIT_ECOMMERCE',
    id,
    updates
});
