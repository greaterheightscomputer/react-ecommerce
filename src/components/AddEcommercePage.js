import React from 'react';

const AddEcommercePage = () => (
    <div>
        This is from my add expense component
    </div>
);

export default AddEcommercePage;