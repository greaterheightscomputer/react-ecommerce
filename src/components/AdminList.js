import React from 'react';
import { connect } from 'react-redux';
import AdminListItem from './AdminListItem';
import selectEcommerce from '../selectors/ecommerces';

export const AdminList = (props) => (
    <div>
        <h1>Admin List</h1>
        {
            props.ecommerces.length === 0 ? (
                <p> No Products </p>
            ) : (
                props.ecommerces.map((ecommerce) =>{
                    return <AdminListItem key={ecommerce.id} {...ecommerce} />
                })
            )
        }                
    </div>
);

const mapStateToProps = (state) => {
    return {
        ecommerces: selectEcommerce(state.ecommerces, state.filters)        
    };
}

export default connect(mapStateToProps)(AdminList);