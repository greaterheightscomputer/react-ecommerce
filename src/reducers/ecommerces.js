const ecommerceReducerDefaultState = [];
const ecommerceReducer = (state = ecommerceReducerDefaultState, action) => {
    switch (action.type) {
        case 'ADD_ECOMMERCE':            
            return [...state, action.ecommerce];   
        case 'REMOVE_ECOMMERCE':            
            return state.filter(({ id }) => id !== action.id); 
        case 'EDIT_ECOMMERCE':
            return state.map((ecommerce) => {
                if (ecommerce.id === action.id){
                    return {...ecommerce, ...action.updates};
                }else {
                    return ecommerce;
                }
            });              
        default: return state;        
    }
};

export default ecommerceReducer;