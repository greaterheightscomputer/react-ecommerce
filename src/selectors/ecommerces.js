import moment from "moment";

//Get visible ecommerce -> this function will match the ecommerce array object with the filters object
export default (ecommerces, { text, sortBy, itemType,  categoryType, startDate, endDate }) => {
    return ecommerces.filter((ecommerce) => {  //filter() allow us to return a subset of all of the array expenses.
        // const startDateMatch = typeof startDate !== 'number' || ecommerce.createdAt >= startDate;
        // const endDateMatch = typeof endDate !== 'number' || ecommerce.createdAt <= endDate;
        const createdAtMoment = moment(ecommerce.createdAt);
        const startDateMatch = startDate ? startDate.isSameOrBefore(createdAtMoment, 'day') : true;
        const endDateMatch = endDate ? endDate.isSameOrAfter(createdAtMoment, 'day') : true;
        const textMatch = ecommerce.description.toLowerCase().includes(text.toLowerCase());
        const categoryMatch = ecommerce.category.includes(categoryType);
        const itemMatch = ecommerce.item.includes(itemType);
        
        return startDateMatch && endDateMatch && textMatch && categoryMatch && itemMatch;
    }).sort((a, b) =>{    //compare function. sort() get called on a return subset array. if a < b return -1 else if a > b return 1  else a === b return 0
          if(sortBy === 'date') {
              return a.createdAt < b.createdAt ? 1 : -1;   // the truthy value is 1 which means b is more recent than a 
        }                                                  
        if(sortBy === 'amount') {
            return a.amount < b.amount ? 1 : -1;
      }
    });
};

