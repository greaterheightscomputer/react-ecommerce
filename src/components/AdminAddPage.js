import React from 'react';
import { connect } from 'react-redux';
import AdminForm from './AdminForm';
import { startAddEcommerce } from '../actions/ecommerces';

export class AdminAddPage extends React.Component {
    onSubmit = (ecommerce) => {
        // console.log(ecommerce)
        // props.dispatch(addEcommerce(ecommerce));
        this.props.startAddEcommerce(ecommerce);
        this.props.history.push('/admin_dashboard');
    };
    render() {
        return (
            <div className="list-body">
                <div className="page-admin-header">
                    <div className="content-container">
                        <h1 className="page-admin-header__title"> Add Product</h1>
                    </div>
                </div>                
                <div className="content-container">
                    <AdminForm
                        onSubmit={this.onSubmit}
                    />
                </div>
            </div>
        );
    };
};

const mapDispatchToProps = (dispatch) => ({
    startAddEcommerce: (ecommerce) => dispatch(startAddEcommerce(ecommerce))
});

export default connect(undefined, mapDispatchToProps)(AdminAddPage);