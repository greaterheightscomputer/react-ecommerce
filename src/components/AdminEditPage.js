import React from 'react';
import { connect } from 'react-redux';
import AdminForm from './AdminForm';
import { editEcommerce, removeEcommerce } from '../actions/ecommerces';

export const AdminEditPage = (props) => {
    // console.log(props)
    return(
    <div>
        <AdminForm 
            ecommerce={props.ecommerce}
            onSubmit={(ecommerce) => {
                props.dispatch(editEcommerce(props.ecommerce.id, ecommerce));
                props.history.push('/admin_dashboard');
                // console.log('updated', ecommerce);
            }}
        />
        <button onClick={() => {
            props.dispatch(removeEcommerce({ id: props.ecommerce.id }));            
            props.history.push('/admin_dashboard');
        }}>Remove</button>
    </div>
    );
};

const mapStateToProps = (state, props) => {
    return {
        ecommerce: state.ecommerces.find((ecommerce) => ecommerce.id === props.match.params.id)
    };
};

export default connect(mapStateToProps)(AdminEditPage);