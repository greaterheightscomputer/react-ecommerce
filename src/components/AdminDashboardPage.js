import React from 'react';
import AdminList from './AdminList';
import AdminListFilters from './AdminListFilters';
import EcommerceSummary from './EcommerceSummary';

const AdminDashboardPage = () => (
    <div>        
        <EcommerceSummary />
        <AdminListFilters />
        <AdminList />        
    </div>
);

export default AdminDashboardPage;