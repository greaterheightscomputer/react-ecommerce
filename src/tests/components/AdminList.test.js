import React from 'react';
import { shallow } from 'enzyme';
import { AdminList } from '../../components/AdminList';
import ecommerces from '../fixtures/ecommerces';

test('should render AdminList with ecommerces', () => {
    const wrapper = shallow(<AdminList ecommerces={ecommerces} />)
    expect(wrapper).toMatchSnapshot();
}); 

test('should render AdminList with empty message', () => {
    const wrapper = shallow(<AdminList ecommerces={[]} />)
    expect(wrapper).toMatchSnapshot();
}); 