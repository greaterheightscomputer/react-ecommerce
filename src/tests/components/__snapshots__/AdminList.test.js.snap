// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AdminList with ecommerces 1`] = `
<div>
  <h1>
    Admin List
  </h1>
  <AdminListItem
    amount={12500}
    category="amen"
    createdAt={0}
    description="men shoe"
    id="1"
    image="shoe.jpeg"
    imageUrl=""
    item="men_footwear"
    key="1"
    stock={100}
  />
  <AdminListItem
    amount={4500}
    category="women"
    createdAt={-432000000}
    description="women shoe"
    id="2"
    image="women_shoe.jpeg"
    imageUrl=""
    item="women_headwear"
    key="2"
    stock={403}
  />
  <AdminListItem
    amount={350}
    category="boy"
    createdAt={432000000}
    description="boy shirt"
    id="3"
    image="shirt_boy.jpeg"
    imageUrl=""
    item="boy_clothe"
    key="3"
    stock={3200}
  />
  <AdminListItem
    amount={2500}
    category="girl"
    createdAt={172800000}
    description="girl head tie"
    id="4"
    image="girl_headtie.jpeg"
    imageUrl=""
    item="girl_headwear"
    key="4"
    stock={17850}
  />
</div>
`;

exports[`should render AdminList with empty message 1`] = `
<div>
  <h1>
    Admin List
  </h1>
  <p>
     No Products 
  </p>
</div>
`;
